src/primitive-field/primitive-field.component.ts

```diff
@@ -71,12 +72,10 @@ export class PrimitiveFieldComponent extends AbstractFieldComponent implements O
   }
 
   commitValueChange() {
-    // Validation
     let errors = this.schemaValidationService.validateValue(this.value, this.schema);
-    if (!errors.length) {
-      this.jsonStoreService.setIn(this.path, this.value);
-    }
-    this.errors = errors;
+    this.jsonStoreService.setIn(this.path, this.value);
+    this.internalErrors = errors;
+    this.appGlobalsService.internalErrors = {[this.pathString]: errors};
```

Comment:
Apart of merging inside this setter there are some custom actions that are not make sense to be inside a custom class that overrides set method. I can name the setter method `extendInternalErrorMap` and convert it to method other than a setter. Would it be more descriptive as by default we extend every time?